File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""THis is a rectangle class"""
class Rectangle:
    """rectangle"""

"""width property
"""
    @property
    def width(self):
        return self.__width

"""width setter
"""
    @width.setter
    def width(self, value):
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

"""height property
"""
    @property
    def height(self):
        return self.__height

"""height setter
"""
    @height.setter
    def height(self, value):
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

"""area
"""
    def area(self):
        return self.__width * self.__height

"""perimeter
"""
    def perimeter(self):
        perimeter = (self.__width * 2) + (self.__height * 2)
        return perimeter

"""the string of object
"""
    def __str__(self):
        str = ''
        if self.__height == 0 or self.__width == 0:
            return str
        for i in range(0, self.__height):
            for j in range(0, self.__width):
                str = str + '#'
            if i != self.__height - 1:
                str = str + '\n'
        return str

"""initialization
"""
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
// Simply an empty class Rectangle with nothing?

class Rectangle {
  constructor (w, h) {
    if (h <= 0 || h <= 0 || w === undefined || h === undefined) {
      return this;
    }
    this.width = w;
    this.height = h;
  }
  print () {
    let p = '';
    let i = 0;
    let j = 0;
    for (i = 0; i < this.height; i++) {
      for (j = 0; j < this.width; j++) {
        p += 'X';
      } console.log(p);
      p = '';
    }
  }
}
module.exports = Rectangle;
